cs-office-365/src/main/java/io/cloudslang/content/office365/utils/PopulateMessageBody.java

```diff
@@ -55,16 +55,24 @@ public static String populateMessageBody(Office365CommonInputs commonInputs, Cre
         final List<Recipient> finalList = new ArrayList<>();
         String[] list = emailInput.split(delimiter);
 
-        for (String item : list) {
-            final EmailAddress emailAddress = new EmailAddress();
-            final Recipient recipient = new Recipient();
-            emailAddress.setAddress(item);
-            recipient.setEmailAddress(emailAddress);
-            finalList.add(recipient);
+        if (!emailInput.contains(delimiter)){
+            setEmailAddress(finalList, emailInput);
+        } else {
+            for (String item : list) {
+                setEmailAddress(finalList, item);
```

Comment:
you should not mutate the `finalList` inside the`setEmailAddress` method. you sould return the recipient and add it to the list here.
eg:
```finalList.add(setEmailAddress(item))```

same goes in the other case of the if

Reply:
@George-Giloan WDYT?

Reply:
😆 I think someone was bored, but also faină! 🥇 

P.S. #SanduCiorba4eva